depends/packages/qt.mk

```diff
@@ -115,6 +110,7 @@ $(package)_config_opts += -no-feature-xml
 $(package)_config_opts_darwin = -no-dbus
 $(package)_config_opts_darwin += -no-opengl
 $(package)_config_opts_darwin += -pch
+$(package)_config_opts_darwin += -device-option QMAKE_MACOSX_DEPLOYMENT_TARGET=$(OSX_MIN_VERSION)
```

Comment:
I think this can actually be dropped.

Reply:
I'm ok with this line as it is.

As Qt 5.12 [supports](https://doc.qt.io/qt-5.12/macos.html#supported-versions) macOS target platforms 10.12, 10.13 and 10.14, and our minimum support macOS is 10.14, it seems reasonable to set [`QMAKE_MACOSX_DEPLOYMENT_TARGET`](https://doc.qt.io/qt-5.12/macos.html#target-platforms) 10.14 rather 10.12.

Reply:
On macOS Sierra 10.12.6 (16G2136) I got nice warnings:

![Screenshot from 2021-03-09 20-07-40](https://user-images.githubusercontent.com/32963518/110517446-89ba9400-8113-11eb-99c6-d9dcf7bc9b1e.png)

![Screenshot from 2021-03-09 20-08-47](https://user-images.githubusercontent.com/32963518/110517452-8cb58480-8113-11eb-9b49-af1487d41056.png)

Reply:
It appears that this line always forces cross-compiling. See #21588.